libraries/AP_NavEKF/AP_Nav_Common.h

```diff
@@ -18,42 +18,46 @@
-        uint16_t horiz_pos_rel      : 1; // 3 - true if the relative horizontal position estimate is valid
-        uint16_t horiz_pos_abs      : 1; // 4 - true if the absolute horizontal position estimate is valid
-        uint16_t vert_pos           : 1; // 5 - true if the vertical position estimate is valid
-        uint16_t terrain_alt        : 1; // 6 - true if the terrain height estimate is valid
-        uint16_t const_pos_mode     : 1; // 7 - true if we are in const position mode
-        uint16_t pred_horiz_pos_rel : 1; // 8 - true if filter expects it can produce a good relative horizontal position estimate - used before takeoff
-        uint16_t pred_horiz_pos_abs : 1; // 9 - true if filter expects it can produce a good absolute horizontal position estimate - used before takeoff
-        uint16_t takeoff_detected   : 1; // 10 - true if optical flow takeoff has been detected
-        uint16_t takeoff            : 1; // 11 - true if filter is compensating for baro errors during takeoff
-        uint16_t touchdown          : 1; // 12 - true if filter is compensating for baro errors during touchdown
-        uint16_t using_gps          : 1; // 13 - true if we are using GPS position
-        uint16_t gps_glitching      : 1; // 14 - true if GPS glitching is affecting navigation accuracy
+        bool attitude           : 1; // 0 - true if attitude estimate is valid
+        bool horiz_vel          : 1; // 1 - true if horizontal velocity estimate is valid
+        bool vert_vel           : 1; // 2 - true if the vertical velocity estimate is valid
+        bool horiz_pos_rel      : 1; // 3 - true if the relative horizontal position estimate is valid
+        bool horiz_pos_abs      : 1; // 4 - true if the absolute horizontal position estimate is valid
+        bool vert_pos           : 1; // 5 - true if the vertical position estimate is valid
+        bool terrain_alt        : 1; // 6 - true if the terrain height estimate is valid
+        bool const_pos_mode     : 1; // 7 - true if we are in const position mode
+        bool pred_horiz_pos_rel : 1; // 8 - true if filter expects it can produce a good relative horizontal position estimate - used before takeoff
+        bool pred_horiz_pos_abs : 1; // 9 - true if filter expects it can produce a good absolute horizontal position estimate - used before takeoff
+        bool takeoff_detected   : 1; // 10 - true if optical flow takeoff has been detected
+        bool takeoff            : 1; // 11 - true if filter is compensating for baro errors during takeoff
+        bool touchdown          : 1; // 12 - true if filter is compensating for baro errors during touchdown
+        bool using_gps          : 1; // 13 - true if we are using GPS position
+        bool gps_glitching      : 1; // 14 - true if GPS glitching is affecting navigation accuracy
     } flags;
     uint16_t value;
 };
 
+static_assert(sizeof(uint16_t) == sizeof(nav_filter_status), "nav_filter_status must be uint16_t");
```

Comment:
So I thought about that orginally, the reason I stuck with `uint16_t` is because places like dataflash depend upon value being a `uint16_t`. Otherwise, the casting they are doing would throw away the upper bits. Happy to change it though if we think the clarity here is better.

Reply:
*shrug*. Mauve, definitely mauve.